modules/KalturaSupport/resources/mw.KCuePoints.js

```diff
@@ -405,6 +409,54 @@
 			}
 			return foundAttr;
 		},
+		/**
+		 * if have same CP earlier - hide current CuePoint
+		 * @param  allCP - array of CP where try to find same CP
+		 * @param  currentCuePointIndex - position from current CP
+		 *
+		 */
+		removeDuplicatedCuePoints:function (allCP, currentCuePointIndex) {
+			var defaultThreshold = this.getThreshold();
+			var currentCP = allCP[currentCuePointIndex];
+			var prevCP = this.getPrevCPWithCorrectType(allCP,currentCuePointIndex);
+			if(prevCP !== false && currentCP){
+				var createdAtDelta = Math.abs(currentCP.createdAt - prevCP.createdAt);
+				var startTimeDelta = Math.abs(currentCP.startTime - prevCP.startTime);
+				var isTheSamePartnerData = currentCP.partnerData === prevCP.partnerData;
+				var isTheSameTags = currentCP.tags === prevCP.tags;
+				if(isTheSamePartnerData && isTheSameTags && (createdAtDelta <= defaultThreshold*1000 || startTimeDelta <= defaultThreshold*1000)){
+					return false;
+				}
+			}
+			return true;
+		},
+		getPrevCPWithCorrectType: function (allCP,currentCuePointIndex) {
+			var prevCP = false;
+			var previousIndex = currentCuePointIndex - 1;
+			var currentCP = allCP[currentCuePointIndex];
+			for(var i = previousIndex; i>=0;i--){
+				var defaultThreshold = this.getThreshold();
```

Comment:
@MaksymKorniiov 
this.getThreshold() is called every iteration, but it never changes. If you have something that you can calculate before the iteration do it outside. With this implementation we call an unnecessary function every iteration.

Reply:
Also, since the Threshold is pretty static and will never change @ runtime - we should calculate this once @ load time and not recalculate it